src/components/prime/googlemap/GMap.tsx

```diff
@@ -0,0 +1,109 @@
+import merge from 'ut-function.merge';
+import * as React from 'react';
+import { GMap as PrimeGMap } from 'primereact/gmap';
+import { Skeleton } from 'primereact/skeleton';
+
+import Context from '../../Text/context';
+
+import { loadGoogleMaps, removeGoogleMaps } from './util';
+import { Props } from './GMap.types';
+
+const defaultOptions = {
+    center: { lat: 42.69641881321328, lng: 23.323133750607305 },
+    zoom: 12,
+    disableDefaultUI: true,
+    clickableIcons: false,
+    gestureHandling: 'cooperative'
+};
+
+const disableUserInteraction = {
+    gestureHandling: 'none'
+};
+
+export const GMap = React.forwardRef<object, Props>(function GMap(props, ref) {
+    if (typeof ref === 'function') ref({});
+    const { options = {}, onChange, value = null, disabled = false } = props;
+    const [googleMapsReady, setGoogleMapsReady] = React.useState(false);
+    const [selectedPosition, setSelectedPosition] = React.useState(value);
+    const [marker, setMarker] = React.useState(null);
+    const map = React.useRef(null);
+    const { configuration: { 'utPrime.GMap': coreConfig = {} } = {}, languageCode = 'bg' } = React.useContext(Context);
+
+    const {key, region, language, ...mapOptions} = React.useMemo(() => {
+        return merge(
+            [
+                {
+                    key: '',
+                    region: 'BG',
+                    language: languageCode
+                },
+                defaultOptions,
+                coreConfig,
+                options,
+                disabled && disableUserInteraction,
+                selectedPosition && { center: selectedPosition }
+            ].filter(Boolean)
+        );
+    }, [options, disabled, selectedPosition, coreConfig, languageCode]);
+
+    React.useEffect(() => {
+        loadGoogleMaps({ language, key, region }, () => {
+            setGoogleMapsReady(true);
+        });
+        return () => {
+            removeGoogleMaps();
+        };
+    }, [language, key, region]);
+
+    React.useEffect(() => {
+        if (!googleMapsReady || !selectedPosition) return;
+        const newMarker = new google.maps.Marker({
```

Comment:
Is there another way to do this, instead of referring to a magic global variable "google" ?

Reply:
not sure, seems like the constructor is only available through the magic var which comes into context once the external google js script has been loaded